src/server/enterprise/server/api_server.go

```diff
@@ -200,16 +201,20 @@ func (a *apiServer) Activate(ctx context.Context, req *ec.ActivateRequest) (resp
 	record := &ec.EnterpriseRecord{License: heartbeatResp.License}
 
 	// If the test heartbeat succeeded, write the state and config to etcd
-	if _, err := col.NewSTM(ctx, a.env.EtcdClient, func(stm col.STM) error {
-		if err := a.configCol.ReadWrite(stm).Put(configKey, &ec.EnterpriseConfig{
+	if err := a.env.TxnEnv.WithWriteContext(ctx, func(txCtx *txncontext.TransactionContext) error {
+		if err := a.configCol.ReadWrite(txCtx.SqlTx).Put(configKey, &ec.EnterpriseConfig{
 			LicenseServer: req.LicenseServer,
 			Id:            req.Id,
 			Secret:        req.Secret,
 		}); err != nil {
 			return err
 		}
-
-		return a.enterpriseTokenCol.ReadWrite(stm).Put(enterpriseTokenKey, record)
+		if _, err := col.NewSTM(ctx, a.env.EtcdClient, func(stm col.STM) error {
```

Comment:
What's going on here.  Do we still need etcd?

Reply:
Background: 
- LS (license service) stores the org's license.
- ES (enterprise service) heartbeats to LS to verify that the license is still valid to continue enabling enterprise
- Here we are moving ES's copy of LS communication info `enterpriseConfig` over to PG
- ES keeps a secondary copy of the license retrieved in the Heartbeat stored in etcd collection called `enterpriseToken`. It's written to infrequently but pulled via `keycache.Cache`. It's valuable to cache it because it's technically accessed at each auth.WhoAmI call.

So whether we need etcd is a good question. My inclination is to leave it as is since it's supposedly more performant - though I don't know for certain it's worth the extra bit of complexity. The enterpriseToken collection also happens to be ephemeral. 

WDYT?

Reply:
I think it's fine to cache stuff in etcd.  Idk about it being more performant though.  We could also cache it locally.